streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingSparkListenerAdapter.scala

```diff
@@ -17,19 +17,16 @@
 
 package org.apache.spark.streaming.scheduler
 
-import java.util.concurrent.atomic.AtomicBoolean
+import org.apache.spark.scheduler.{SparkListener, SparkListenerEvent}
 
-import org.apache.spark.Logging
-import org.apache.spark.util.AsynchronousListenerBus
-
-/** Asynchronously passes StreamingListenerEvents to registered StreamingListeners. */
-private[spark] class StreamingListenerBus
-  extends AsynchronousListenerBus[StreamingListener, StreamingListenerEvent]("StreamingListenerBus")
-  with Logging {
-
-  private val logDroppedEvent = new AtomicBoolean(false)
+/**
+ * A SparkListener adapter for StreamingListener that will dispatch the Streaming events to
+ * the underlying StreamingListener.
+ */
+private[spark] class StreamingSparkListenerAdapter(listener: StreamingListener)
```

Comment:
do we need this class? What if we made `StreamingListener` extend `SparkListener` and override `onOtherEvent` to delegate the streaming events?